pkg/connection/config.go

```diff
@@ -0,0 +1,306 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connection
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"go.uber.org/multierr"
+)
+
+const ActiveConfigPointerFilename = "active_config"
+
+// Flags defines Flags that may be bound to a Configuration. Use like:
+// `cmd.PersistentFlags().AddFlagSet(connection.Flags)`
+var Flags *pflag.FlagSet = createFlagSet()
```

Comment:
This actually seems pretty tricky... Instead of trying to rework it right now, we could potentially change access to the FlagSet via a public accessor for now instead of the public variable... That would at least give us a point of control for future work. Otherwise, I think this will be a significant refactor that should probably also incorporate the separating the config and client concerns mentioned in the other comment - and maybe in a follow up PR?